rollup/da_syncer/da/calldata_blob_source.go

```diff
@@ -8,60 +8,43 @@ import (
 
 	"github.com/scroll-tech/go-ethereum/accounts/abi"
-	"github.com/scroll-tech/go-ethereum/common"
-	"github.com/scroll-tech/go-ethereum/core/types"
 	"github.com/scroll-tech/go-ethereum/ethdb"
-	"github.com/scroll-tech/go-ethereum/log"
 	"github.com/scroll-tech/go-ethereum/rollup/da_syncer/blob_client"
 	"github.com/scroll-tech/go-ethereum/rollup/da_syncer/serrors"
-	"github.com/scroll-tech/go-ethereum/rollup/rollup_sync_service"
+	"github.com/scroll-tech/go-ethereum/rollup/l1"
 )
 
 const (
-	callDataBlobSourceFetchBlockRange  uint64 = 500
-	commitBatchEventName                      = "CommitBatch"
-	revertBatchEventName                      = "RevertBatch"
-	finalizeBatchEventName                    = "FinalizeBatch"
-	commitBatchMethodName                     = "commitBatch"
-	commitBatchWithBlobProofMethodName        = "commitBatchWithBlobProof"
-
-	// the length of method ID at the beginning of transaction data
-	methodIDLength = 4
+	callDataBlobSourceFetchBlockRange uint64 = 500
 )
 
 var (
 	ErrSourceExhausted = errors.New("data source has been exhausted")
 )
 
 type CalldataBlobSource struct {
-	ctx                           context.Context
-	l1Client                      *rollup_sync_service.L1Client
-	blobClient                    blob_client.BlobClient
-	l1height                      uint64
-	scrollChainABI                *abi.ABI
-	l1CommitBatchEventSignature   common.Hash
-	l1RevertBatchEventSignature   common.Hash
-	l1FinalizeBatchEventSignature common.Hash
-	db                            ethdb.Database
+	ctx            context.Context
+	l1Reader       *l1.Reader
+	blobClient     blob_client.BlobClient
+	l1height       uint64
+	scrollChainABI *abi.ABI
+	db             ethdb.Database
 
 	l1Finalized uint64
 }
 
-func NewCalldataBlobSource(ctx context.Context, l1height uint64, l1Client *rollup_sync_service.L1Client, blobClient blob_client.BlobClient, db ethdb.Database) (*CalldataBlobSource, error) {
-	scrollChainABI, err := rollup_sync_service.ScrollChainMetaData.GetAbi()
+func NewCalldataBlobSource(ctx context.Context, l1height uint64, l1Reader *l1.Reader, blobClient blob_client.BlobClient, db ethdb.Database) (*CalldataBlobSource, error) {
+	scrollChainABI, err := l1.ScrollChainMetaData.GetAbi()
 	if err != nil {
 		return nil, fmt.Errorf("failed to get scroll chain abi: %w", err)
 	}
 	return &CalldataBlobSource{
-		ctx:                           ctx,
-		l1Client:                      l1Client,
-		blobClient:                    blobClient,
-		l1height:                      l1height,
-		scrollChainABI:                scrollChainABI,
-		l1CommitBatchEventSignature:   scrollChainABI.Events[commitBatchEventName].ID,
-		l1RevertBatchEventSignature:   scrollChainABI.Events[revertBatchEventName].ID,
-		l1FinalizeBatchEventSignature: scrollChainABI.Events[finalizeBatchEventName].ID,
-		db:                            db,
+		ctx:            ctx,
+		l1Reader:       l1Reader,
+		blobClient:     blobClient,
+		l1height:       l1height,
```

Comment:
Would `l1Height`  be more consistent with the namings? feel free to change it or not since it's a code migration pr.